shell/mput.go

```diff
@@ -0,0 +1,195 @@
+package shell
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/abiosoft/ishell"
+	"github.com/juruen/rmapi/util"
+)
+
+func mputCmd(ctx *ShellCtxt) *ishell.Cmd {
+	return &ishell.Cmd{
+		Name:      "mput",
+		Help:      "recursively copy local files to remote directory",
+		Completer: createFsEntryCompleter(),
+		Func: func(c *ishell.Context) {
+
+			argsLen := len(c.Args)
+
+			if argsLen == 0 {
+				c.Err(errors.New(("missing destination dir")))
+				return
+			}
+
+			if argsLen > 1 {
+				c.Err(errors.New(("too many arguments for command mput")))
+				return
+			}
+
+			// Past this point, the number of arguments is 1.
+
+			node, err := ctx.api.Filetree.NodeByPath(c.Args[0], ctx.node)
+
+			if err != nil || node.IsFile() {
+				c.Err(errors.New("remote directory does not exist"))
+				return
+			}
+
+			path, err := ctx.api.Filetree.NodeToPath(node)
+
+			if err != nil || node.IsFile() {
+				c.Err(errors.New("remote directory does not exist"))
+				return
+			}
+
+			// Back up current remote location.
+			currCtxPath := ctx.path
+			currCtxNode := ctx.node
+			// Change to requested directory.
+			ctx.path = path
+			ctx.node = node
+
+			putFilesAndDirs(ctx, c, "./", 0)
+
+			// Reset.
+			ctx.path = currCtxPath
+			ctx.node = currCtxNode
+		},
+	}
+}
+
+// Checks whether the file has a pdf or epub extension.
+// Input -> Valid file name.
+// Returns -> true if the file is a pdf or epub
+//		   -> false otherwise
+func checkFileType(fName string) bool {
+	return (strings.Contains(fName, ".pdf") ||
+		strings.Contains(fName, ".epub"))
+}
+
+// Print the required number of spaces
+func treeFormat(pC *ishell.Context, num int) {
+
+	if num != 0 {
+		pC.Printf("│")
+	} else {
+		pC.Printf("├")
+	}
+
+	for i := 0; i < 2*num; i++ {
+		pC.Printf(" ")
+	}
+
+	if num != 0 {
+		pC.Printf("  └── ")
+	} else {
+		pC.Printf("── ")
+	}
+}
+
+func putFilesAndDirs(pCtx *ShellCtxt, pC *ishell.Context, localDir string, depth int) bool {
+
+	if localDir == "./" {
+		pC.Println(".")
+	}
+
+	os.Chdir(localDir) // Change to the local source directory.
+
+	wd, _ := os.Getwd()
+	// pC.Println("DEBUG: changing to directory", wd)
```

Comment:
Can we remove this line? Or if you still think it might be useful for future debugging you can change it to `log.Trace` which won't log anything by default.

Reply:
Removed.